mlir/utils/performance/ck-benchmark-driver/ck-atten-benchmark-driver.cpp

```diff
@@ -0,0 +1,475 @@
+  const auto opPtrs = Operation::getInstances();
+
+  if (opPtrs.empty()) {
+    throw std::runtime_error("failed to find any config for the given options");
+  } else {
+    std::cout << "found " << opPtrs.size() << " `AttenOp` candidates"
+              << std::endl;
+  }
+
+  Data data{options};
+
+  std::string bestKernelName{"none"};
+  int bestKernelId{-1};
+  float minTime{std::numeric_limits<float>::max()};
+
+  for (size_t idx = 0; idx < opPtrs.size(); ++idx) {
+    auto &opPtr = opPtrs[idx];
+    auto argumentPtr = Operation::makeArgs(opPtr, data, options);
+    if (opPtr->IsSupportedArgument(argumentPtr.get())) {
+      std::string opName = opPtr->GetTypeString();
+      auto invokerPtr = opPtr->MakeInvokerPointer();
+      StreamConfig streamConfig{nullptr, true};
+      float avgTime = invokerPtr->Run(argumentPtr.get(), streamConfig);
+
+      avgTime /= static_cast<float>(streamConfig.nrepeat_);
+      if (avgTime < minTime) {
+        minTime = avgTime;
+        bestKernelName = opPtr->GetTypeString();
+        bestKernelId = static_cast<int>(idx);
+      }
+      if (options.verbose) {
+        std::cout << idx << ") under test\n\t" << opPtr->GetTypeString()
+                  << std::endl;
+      }
+    } else {
+      if (options.verbose) {
+        std::cout << idx << ") arguments are NOT supported for\n\t"
+                  << opPtr->GetTypeString() << std::endl;
+      }
+    }
+  }
+
+  if (bestKernelId >= 0) {
+    const auto flop = Operation::getFlopCount(options);
+    const float tflops = static_cast<float>(flop) / 1.E9 / minTime;
+
+    std::cout << "Best kernel report:"
+              << "\nkernel time: " << minTime << "\nkernel ID: " << bestKernelId
+              << "\nTFlops: " << tflops << "\nkernel name:\n\t"
+              << bestKernelName << std::endl;
+  } else {
+    throw std::runtime_error(
+        "arguments are not supported by all AttentionOp candidates");
+  }
+}
+
+struct Selector {
+  void dispatch(Options &options) {
+    try {
+      switch (options.elementType) {
+      case benchmark::DataType::F32: {
+        selectOperation<float>(options);
+        break;
+      }
+      case benchmark::DataType::F16: {
+        selectOperation<ck::half_t>(options);
+        break;
+      }
+      case benchmark::DataType::I8: {
+        selectOperation<int8_t>(options);
+        break;
+      }
+      default: {
+        throw std::runtime_error("provided unsopported data element type");
+      }
+      };
+    } catch (std::runtime_error &err) {
+      throw err;
+    }
+  }
+
+  template <typename Type>
+  void selectOperation(Options &options) {
+    if (options.hasAttnBias) {
+      runBenchmark<AttentionWithBias<Type>>(options);
+    } else {
+      runBenchmark<AttentionWithoutBias<Type>>(options);
+    }
+  }
+};
+
+int main(int argc, char *argv[]) {
+  Options options{};
+  CLI::App app{"elementwise example with multi-dim block/grid"};
+
+  app.add_option("--seq_len", options.sequenceLength,
```

Comment:
I think we need these to match cross attention args which is the super set of self & cross attentions.

Reply:
Done. Now the command line interface is compatible with what we have

https://github.com/ROCm/rocMLIR/blob/f070321e0d4d807b2fccfe748c20347196768bbd/mlir/utils/performance/ck-benchmark-driver/ck-atten-benchmark-driver.cpp#L463-L472